HLTrigger/JetMET/src/HLTJetEtaTopologyFilter.cc

```diff
@@ -0,0 +1,104 @@
+/** \class HLTJetEtaTopologyFilter
+ *
+ *
+ *  \author Tomasz Fruboes
+ *     based on HLTDiJetAveFilter
+ */
+
+#include "HLTrigger/JetMET/interface/HLTJetEtaTopologyFilter.h"
+
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "HLTrigger/HLTcore/interface/defaultModuleLabel.h"
+
+#include<typeinfo>
+
+//
+// constructors and destructor
+//
+template<typename T>
+HLTJetEtaTopologyFilter<T>::HLTJetEtaTopologyFilter(const edm::ParameterSet& iConfig) : HLTFilter(iConfig),
+    inputJetTag_ (iConfig.template getParameter< edm::InputTag > ("inputJetTag")),
+    minPtJet_    (iConfig.template getParameter<double> ("minPtJet")),
+    //minPtJet3_   (iConfig.template getParameter<double> ("minPtJet3")),
+    jetEtaMin_     (iConfig.template getParameter<double> ("minJetEta")),
+    jetEtaMax_     (iConfig.template getParameter<double> ("maxJetEta")),
+    applyAbsToJet_ (iConfig.template getParameter<bool> ("applyAbsToJet")),
+
+    triggerType_ (iConfig.template getParameter<int> ("triggerType"))
+{
+    m_theJetToken = consumes<std::vector<T>>(inputJetTag_);
```

Comment:
... by moving this line up into the initialiser list via
```
m_theJetToken_(consumes<std::vector<T>>(inputJetTag_)),
```
after the line where inputJetTag_ is initialised.